### ISSUE TITLE
No tests fail if `PyiAwareFlakesChecker.ASSIGN` is deleted

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#374]
If I delete the `PyiAwareFlakesChecker.ASSIGN` method from our code, none of our tests fail. However, deleting the method causes a number of undesirable false positives when running flake8 on typeshed (if F821 is reenabled in typeshed's `.flake8` file). For example, flake8 starts complaining about forward references in `TypeVar` bounds, such as this:

```py
_CT = TypeVar("_CT", bound=_CData)
class _CData: ...
```

As a matter of style, it's nice to have all the TypeVars defined at the top of the file; and forward references are actually often unavoidable in TypeVar bounds. So this is a false positive that we don't want.

This points to a hole in our test coverage: we should add tests to cover cases like these.
